### DISCUSSION TITLE
Home Assistant add-on

### DISCUSSION THREAD

**Discussion by Duetting** [Akkudoktor-EOS/EOS#294] [General]
Hi,
I tested the last few days integrating the EOS into my Home Assistant instance without all the Node-Red stuff.
To make it the usage even easier for me, I also created an add-on repo with this repo as submodule. 
The logo i used was just created by Copilot.
I hope i didn't do anything wrong here or made something that's already existing.

I could provide more information about all the intermediate sensors that i used to pass the information to the optimize endpoint, but i fear they are very specific for my setup.

The repo can also be moved somewhere else, I just wanted to get it working before.

Before i forget it, here is the link to the repo: https://github.com/Duetting/ha_eos_addon

The log messages from the s6 service still doesn't seem to work reliably, but the service works for me.

**Comment by Lasall** [comment]
Pretty cool integration! We aim to provide docker images that can then just be added in config.yaml (see #277). Then the HA addon could be integrated just like https://github.com/evcc-io/hassio-addon . The goal is to have a seemless integration into the HA ecosystem (UI). One example could be https://github.com/marq24/ha-evcc .

As a first step your integration shown in https://github.com/Duetting/ha_eos_addon/blob/main/eos/DOCS.md is quite neat and we could definitely start from there as a very first step of HA support (documentation on how to use EOS with direct API calls).

Edit: An intermediate step before a true HA-integration would be communication over MQTT, which simplifies a little bit compared to direct API calls.

**Comment by Duetting** [comment]
I used my own dockerfile, cause it was easier for me to use one of the home assistant base images.
To be able to use it as add-on in home assistant, the image should have bashio and tempio installed and use s6.
Maybe you should consider at least s6, for your docker image and when there is a stable docker image available, the home assistant add-on dockerfile should only add bashio and maybe tempio and adapt the s6 scripts.
Providing all the necessary data for the optimize method in a home assistant friendly and customizable way, will be difficult I fear. Only the home assistant api provides a guaranteed way of accessing historical values for sensors, e.g.
An integration could easily provide the temperature forecast and soc values. But making the consumption, PV forecast and dynamic price values configurable, will be difficult imho.

**Comment by stheid** [comment]
Does this really have to be an addon or can it be an integration? I have limited experience with integrations, but none with addons

**Comment by MatzeBu** [comment]
I'm not sure if there is a way to combinate this addon and EOS connect 
https://github.com/ohAnd/EOS_connect

I tried it, both systems seem to work seperate. But at the eos connect Dashboard Is only "waiting for first data..."

Perhaps the problem is also because EOS connect ist at the moment for Fronius inverter. I have sungrow.

**Reply by Duetting** [reply]
I only did take a short look at EOS connect, but it's using a different philosophy (maybe a better one) than the EOS addon.
I didn't want to add anything more to the addon, than just making the EOS system available directly in home assistant. And the "rest" needs to be handled in home assistant with the help of a bunch of HACS integrations, helpers, SQL entities and automations. That produces a very flexible solution at the cost of a lot of additional work. I also don't own a Fronius inverter, mine is from Kostal.

**Reply by Duetting** [reply]
And my personal opinion about accessing and sending entity values in a "home assistant" way is by using an integration, not via web requests sent from an addon.
And I already have a (probably complicated looking) setup of automations controlling my home. So if anything breaks it's my own fault, but I can also fix it myself, cause I know where to look, at least up to now.

**Reply by ohAnd** [reply]
Hi @MatzeBu ,

both addons are seperated from each other.

The *EOS HA addon* of @Duetting provide the service of EOS and with this, there is an instance of the optimisation system available what can be used to configure the model and request the optimization with own provided data via the provided API of EOS.

*EOS connect* (what the name tries to express ;-) ) connect your data provided by different sources with an already existing EOS instance and retrives the optimisation result and controls (at this time) a fronius invetrer/ battery with that information . Means it expects a running EOS instance - it could be EOS HA Addon or a running instance e.g. on another docker system in your network or externally.

_(as you mentioned Fronius vs. sungrow - yes, currently only fronius is directly supported - but you can switch in config to default at inverter and EOS connect runs without a direct controlling and provide the target control data via a small API to use that info to control e.g. sungrow via existing HA integrations)_

Conclusion: if there is no running EOS instance - EOS connect have no work ;-)

Btw: @Duetting thanks for your work with the standalone EOS HA addon to have this single instance of EOS easily runnable on HA OS. (with a very small trigger - maybe there is a chance to update to the most recent EOS version within the AddOn ;-) )

**Reply by thecem** [reply]
@ohAnd have a Look in my fork of Duetting EOS, so you can use the actual main version.

**Reply by WolfImBusch** [reply]
Starting from version 0.203.0 evcc has extended its batterie api (https://github.com/evcc-io/evcc/pull/20455). So maybe there is a good chance for eos_connect to control other inverters and therefore batteries beside Fronius (https://github.com/ohAnd/EOS_connect/issues/12) ... Up to now the evcc docu has unfortunately not been updated ...

**Reply by MatzeBu** [reply]
> Hi @MatzeBu ,
> 
> both addons are seperated from each other.
> 
> The _EOS HA addon_ of @Duetting provide the service of EOS and with this, there is an instance of the optimisation system available what can be used to configure the model and request the optimization with own provided data via the provided API of EOS.
> 
> _EOS connect_ (what the name tries to express ;-) ) connect your data provided by different sources with an already existing EOS instance and retrives the optimisation result and controls (at this time) a fronius invetrer/ battery with that information . Means it expects a running EOS instance - it could be EOS HA Addon or a running instance e.g. on another docker system in your network or externally.
> 
> _(as you mentioned Fronius vs. sungrow - yes, currently only fronius is directly supported - but you can switch in config to default at inverter and EOS connect runs without a direct controlling and provide the target control data via a small API to use that info to control e.g. sungrow via existing HA integrations)_
> 
> Conclusion: if there is no running EOS instance - EOS connect have no work ;-)
> 
> Btw: @Duetting thanks for your work with the standalone EOS HA addon to have this single instance of EOS easily runnable on HA OS. (with a very small trigger - maybe there is a chance to update to the most recent EOS version within the AddOn ;-) )

Thanks for your answer!
So I thing it should work, when I let EOS Connect with my eos in HA from duetting. 
But EOS connect say: “waiting for finishing first optimization run“

I'm not sure: which configuration is necessary in EOS?

Eos connect configuration is complete.

**Reply by WolfImBusch** [reply]
Hi @MatzeBu ,

maybe I can help out (Beside that I propose to discuss that in the eos_connect repository).

First coming back to your question: The **EOS** default config file should be sufficient in order to start EOS. Further configuration of EOS is not needed. All necessary information to start EOS via EOS_connect (forecast, temperature, ...) are collected via EOS_connect.

Beside that: the first run may take - depending on your hardware EOS is running - several minutes (and may look strange). On my Synology DS220+ approximately 4 minutes. 

I ask you to check the following points

1) please check if EOS is up an running. I propose 
`http://<IP-addres>:<port>/docs ` 
`http://<IP-addres>:<port>/v1/health`
You should see a documentation page for the API and something like "healthy"

2) Please check if (**this** EOS) IP address is the same as in **EOS_connect** "config.yaml" file and increase the "timeout" in a first step to **600s**
```
eos:
  server: <IP-addres>  # Default EOS server address
  port: <port>  # Default port for EOS server
  timeout: 600 # timeout for EOS optimize request in seconds
```

If this does not help I ask you to attach a log file.

Good luck!

**Reply by MatzeBu** [reply]
Thank you very much for your help!
the docs Page work.
v1/health show: 404 Not Found

It runs on a Raspberry 4 with 8gb


Edit:
Horizont was missing. Now it works! Thank you!

**Reply by Duetting** [reply]
Hi @WolfImBusch,
I should not have commented on EOS connect, before taking a deeper look...
Anyway the EOS addon is now available in a local build variant, which allows to always use the newest version of the EOS system. The only downsides to that variant are:
1. Having to fix breaking changes in the EOS system independent from the addon version. But you can choose yourself when and if you want to rebuild...
2. The creation of the docker image locally can take a lot of time, especially on slower hardware like a raspberry pi.
3. The image is a bit bigger, cause of the additionally needed git client and the complete EOS repo inside of it.

**Reply by ohAnd** [reply]
> Thank you very much for your help! the docs Page work. v1/health show: 404 Not Found
> 
> It runs on a Raspberry 4 with 8gb
> 
> Here my logs:
> 
> [logs-1.txt](https://github.com/user-attachments/files/19829279/logs-1.txt) [logs.txt](https://github.com/user-attachments/files/19829280/logs.txt)

Hi @MatzeBu ,

maybe it's better we shifting with your questions to https://github.com/ohAnd/EOS_connect/issues (@WolfImBusch already mentioned) - your topic isn't directly related to EOS or EOS HA addon from @Duetting .

shortly to your points:
1. if the doc page works and v1/health not reachable then you're driving the older version of EOS - it's no problem for EOS connect - it will be detected and handled.
2. To your logs - it's also showing a misconfiguration for the pv forecast 'horizont' - so check your config regarding the description/ Config readme and try again.

And finally let's step out to eos connect isssues - to stay focus on EOS topics here... ;-)
